src/dinit-monitor.cc

```diff
@@ -480,3 +558,95 @@ static bool load_service(int socknum, cpbuffer_t &rbuffer, const char *name, han
 
     return true;
 }
+
+static void request_environ(int socknum, cpbuffer_t &rbuffer)
+{
+    char c = (char)cp_cmd::LISTENENV;
+    write_all_x(socknum, &c, 1);
+
+    wait_for_reply(rbuffer, socknum);
+
+    cp_rply reply_pkt_h = (cp_rply)rbuffer[0];
+    if (reply_pkt_h != cp_rply::ACK) {
+        throw dinit_protocol_error();
+    }
+    rbuffer.consume(1);
+}
+
+static size_t get_allenv(int socknum, cpbuffer_t &rbuffer)
```

Comment:
So this reads and processes the packet header and leaves the body of the reply pending? Could you add a comment explaining that

Reply:
yes, added